roles/edpm_nova/tasks/configure_compute.yml

```diff
@@ -0,0 +1,67 @@
+---
+# Copyright 2024 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Configure compute configs
+  block:
+    - when: edpm_nova_cpu_shared_set|string
+      name: Set CPU shared set
+      community.general.ini_file:
+        path: "{{ edpm_nova_config_dest }}/03_compute_config.conf"
+        setype: container_file_t
+        mode: "0644"
+        create: true
+        section: "compute"
+        option: "cpu_shared_set"
+        value: "{{ edpm_nova_cpu_shared_set }}"
+
+    - when: edpm_nova_cpu_dedicated_set|string
+      name: Set CPU dedicated set
+      community.general.ini_file:
+        path: "{{ edpm_nova_config_dest }}/03_compute_config.conf"
+        setype: container_file_t
+        mode: "0644"
+        create: true
+        section: "compute"
+        option: "cpu_dedicated_set"
+        value: "{{ edpm_nova_cpu_dedicated_set }}"
+
+    - when: edpm_nova_reserved_host_memory|int > 0
+      name: Set reserved host memory
+      community.general.ini_file:
+        path: "{{ edpm_nova_config_dest }}/03_compute_config.conf"
+        setype: container_file_t
+        mode: "0644"
+        create: true
+        section: "compute"
+        option: "reserved_host_memory_mb"
+        value: "{{ edpm_nova_reserved_host_memory }}"
+  notify:
+    - Restart nova
+
+- name: Configure SRIOV PCI Passthrough
+  when: (edpm_nova_pci_passthrough | list | length) > 0
+  community.general.ini_file:
+    path: "{{ edpm_nova_config_dest }}/04_sriov_config.conf"
+    setype: container_file_t
+    mode: "0644"
+    create: true
+    section: "pci"
+    option: "passthrough_whitelist"
+    value: "{{ item | to_json }}"
+    exclusive: false
+  loop: "{{ edpm_nova_pci_passthrough | list }}"
+  notify:
+    - Restart nova
```

Comment:
the current proposal is also not correct.
passthrough_whitelist is deprecated and we had a company wide goal to remove all reference to blacklist/whitelist form our product device_spec is the replacement name.

https://docs.openstack.org/nova/latest/configuration/config.html#pci.device_spec and passthrough_whitelist were both multi opt values meaning that 
the behavior is different from in format in that each time is merged.

```
- name: Configure SRIOV PCI Passthrough
  when: (edpm_nova_pci_passthrough | list | length) > 0
  community.general.ini_file:
    path: "{{ edpm_nova_config_dest }}/04_sriov_config.conf"
    setype: container_file_t
    mode: "0644"
    create: true
    section: "pci"
    option: "passthrough_whitelist"
    value: "{{ item | to_json }}"
    exclusive: false
  loop: "{{ edpm_nova_pci_passthrough | list }}"
  notify:
    - Restart nova
 ```

 exclusive: false

changes the behaviour somewhat
```
If set to false, only the specified [value](https://docs.ansible.com/ansible/latest/collections/community/general/ini_file_module.html#ansible-collections-community-general-ini-file-module-parameter-value)/[values](https://docs.ansible.com/ansible/latest/collections/community/general/ini_file_module.html#ansible-collections-community-general-ini-file-module-parameter-values) are added when [state=present](https://docs.ansible.com/ansible/latest/collections/community/general/ini_file_module.html#ansible-collections-community-general-ini-file-module-parameter-state), or removed when [state=absent](https://docs.ansible.com/ansible/latest/collections/community/general/ini_file_module.html#ansible-collections-community-general-ini-file-module-parameter-state), and existing ones are not modified.
```
the default state is present so this would just addtivly append to the file across multiple iterations if we change the input values.